tests/Unit/Domain/Services/AuthenticationExceptionTest.php

```diff
@@ -21,10 +21,31 @@
  */
 final class AuthenticationExceptionTest extends Framework\TestCase
 {
+    public function testIsFinal()
+    {
+        $reflection = new \ReflectionClass(AuthenticationException::class);
+        $this->assertTrue($reflection->isFinal());
+    }
+
     public function testIsException()
     {
         $exception = new AuthenticationException();
 
         $this->assertInstanceOf(\Exception::class, $exception);
     }
+
+    public function testIsRuntimeException()
+    {
+        $exception = new AuthenticationException();
+
+        $this->assertInstanceOf(\RuntimeException::class, $exception);
+    }
+
+    public function testLoginFailureHasCorrectMessage()
+    {
+        $exception = AuthenticationException::loginFailure();
```

Comment:
How about asserting that `AuthenticationException:: loginFailure()` returns an `AuthenticationException` as well?

Reply:
No problem